### DISCUSSION TITLE
Cobbler3.2.2 pxe guides to install centos7.6, but the PXE network menu cannot be displayed

### DISCUSSION THREAD

**Discussion by xjr1** [cobbler/cobbler#3284] [Version specific questions]
The version cobbler3.2.2 installs centos7.6, the legacy boot cannot display the PXE network menu; the UEFI boot can display the pxe network menu normally;
The virtual machine operating system built by vmware esxi: Rocky8.5 as the PXE cobbler server；

![fc7561c2b31e938d9d096dbff6f0042](https://user-images.githubusercontent.com/93509872/204727780-44e501e4-6ec1-4e4f-9d42-3104fc4dd1dc.png)

[root@bogon ~]#  tail -f /var/log/messages  
Through the log, it is found that it cannot be loaded normally  grub/grub.cfg、grub/local_legacy.cfg、grub/x86_64_menu_items.cfg
![7af912df8b1fa7be3b6907fa4f05c7c](https://user-images.githubusercontent.com/93509872/204727858-6a74b85c-71d5-4d40-a86a-f9d1e5466ccb.png)

[root@bogon ~]# systemctl status dhcpd
[root@bogon ~]# systemctl status tftp

![017c40306531fb10129576ba9bfb618](https://user-images.githubusercontent.com/93509872/204728463-cec2e1a6-67d6-41d2-89a8-0caa6454e0b5.png)

[root@bogon ~]# vim /etc/cobbler/dhcp.template
![fcf01712d6780e30124e563fd9e6b1e](https://user-images.githubusercontent.com/93509872/204728563-9f2c3d0c-a980-42cc-83d3-6aa792260221.png)

[root@bogon ~]# cobbler sync          # execute normally

[root@bogon ~]# ls /var/lib/tftpboot/grub   
![image](https://user-images.githubusercontent.com/93509872/204729449-654edf7b-c925-4f7e-abe1-5a335273269f.png)

**Comment by SchoolGuy** [comment]
So as far as I see in the log of the TFTP Server, the GRUB binary `grub.0` is transferred, thus I would expect an error message by GRUB on the client. However on your screenshot it is clearly visible that this is not the case. I must say that ad-hoc I have no clue how to debug this. If the EFI menu is working, is there a reason why this is not used?

**Reply by xjr1** [reply]
Client, timeout prompted.  
grub>

![image](https://user-images.githubusercontent.com/93509872/204746976-21899c88-f049-4569-bd0e-17edff35c826.png)

**Reply by SchoolGuy** [reply]
Ahh that makes more sense now. Okay if this is the case there may be an error in the requested GRUB configuration file name. Can you see on the TFTP server what files the client is requesting vs the files that are present?

**Reply by xjr1** [reply]
Hello, I checked that the file requested by the client on the tftp server is the same as /var/lib/tftpboot/grub.0 .

**Reply by SchoolGuy** [reply]
Okay it seems I sas not precise enough. Normally GRUB requests a configuration file after grub.0 was requested. The name of the file may vary depending on your Cobbler configuration. It contains either the MAC address in its name of it is named grub.cfg.

**Reply by xjr1** [reply]
My client is a virtual machine, and the tftp server cannot request the /var/lib/tftpboot/grub/grub.cfg file, which makes it impossible to go to the local network menu interface. Today, I can load it to the local network menu normally with the physical machine client;

**Reply by SchoolGuy** [reply]
But why does this then work when UEFI is being used?

**Reply by xjr1** [reply]
Hello, thank you for your reply, I also don't understand, UEFI can do it, traditional boot can't;

**Reply by SchoolGuy** [reply]
Is an already installed virtual machine able to access the files via TFTP?

**Reply by xjr1** [reply]
I can't, I'm sorry, I just saw you;

**Reply by SchoolGuy** [reply]
So if another installed machine cannot reach the Cobbler TFTP then you have a networking problem. Your installer needs to be able to reach the TFTP and HTTP server Cobbler is serving.
